CHANGELOG.md

```diff
@@ -37,6 +37,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### New features
 
+* `jj squash` now accepts a `--keep-empty` option to keep empty commits.
+  This option is implied by `--interactive` and specifying paths.
```

Comment:
I am not sure about this (still thinking a bit), but if we *are* changing the behavior in this way, it should be mentioned in "breaking changes".

Reply:
I also wonder whether we should print a warning, either temporarily or permanently, if the commit is emptied but `--interactive` or paths are specified. Something like "The commit ... is now empty; use `jj abandon ...` to abandon it".

Reply:
I added it to the breaking changes. I'm not opposed to adding a warning, but I'd like to hear other people's thoughts first.

Reply:
I think this breaking change is a good idea, a person specifying paths is much more likely to want to `--keep-emptied`. My only worry is surprising people who relied on the old behavior or making their use-cases harder.

To avoid making use-cases harder, we could have an `--abandon-emptied` option, but in my mind the warning would make it unnecessary. It's easy enough to `jj abandon` the unnecessary commit; the only difficulty is that without a warning you'd have to check whether it is actually empty. (I'm also not sure how common such use-cases are, but I can imagine wanting both a squash and information on whether that nullified a commit when doing something like `jj squash cli/*` )

I don't think a warning is absolutely essential, though I like it more the more I think about it. OTOH, I know that @yuja and @martinvonz sometimes dislike such warnings.

Reply:
Marking as resolved since I'm no longer making this change a breaking one.